annotations-runtime/shared/src/main/java/com/github/dynamicextensionsalfresco/metrics/Timer.java

```diff
@@ -0,0 +1,31 @@
+package com.github.dynamicextensionsalfresco.metrics;
+
+import java.util.function.Supplier;
+
+/**
+ * register timing information during a transaction and report after commit (reporting dependant of implementation)
+ *
+ * @author Laurent Van der Linden
+ */
+public interface Timer {
+
+    void start(String label);
+
+    void stop();
+
+    boolean isEnabled();
+
+    default <T> T time(Supplier<String> labelProvider, Supplier<T> operation) {
```

Comment:
The best I could invent to replace the existing inline function. Feedback welcome.